subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/incremental/recomp/SourceFileClassNameConverter.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.tasks.compile.incremental.recomp;
+
+import java.io.File;
+import java.util.Collection;
+import java.util.Optional;
+
+public interface SourceFileClassNameConverter {
+    Collection<String> getClassNames(File sourceClassFile);
+
+    Optional<File> getFile(String fqcn);
```

Comment:
I think this method should only be in `GroovySourceFileClassNameConverter` and not in the general interface, since it is not used from Java.

Reply:
Done.